rules/aws_provider_missing_tags.go

```diff
@@ -0,0 +1,160 @@
+	return &AwsProviderMissingTagsRule{}
+}
+
+// Name returns the rule name
+func (r *AwsProviderMissingTagsRule) Name() string {
+	return "aws_provider_missing_tags"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *AwsProviderMissingTagsRule) Enabled() bool {
+	return false
+}
+
+// Severity returns the rule severity
+func (r *AwsProviderMissingTagsRule) Severity() tflint.Severity {
+	return tflint.NOTICE
+}
+
+// Link returns the rule reference link
+func (r *AwsProviderMissingTagsRule) Link() string {
+	return project.ReferenceLink(r.Name())
+}
+
+// Check checks providers for missing tags
+func (r *AwsProviderMissingTagsRule) Check(runner tflint.Runner) error {
+	config := awsProviderTagsRuleConfig{}
+	if err := runner.DecodeRuleConfig(r.Name(), &config); err != nil {
+		return err
+	}
+
+	providerSchema := &hclext.BodySchema{
+		Attributes: []hclext.AttributeSchema{
+			{
+				Name:     "alias",
+				Required: false,
+			},
+		},
+		Blocks: []hclext.BlockSchema{
+			{
+				Type: providerDefaultTagsBlockName,
+				Body: &hclext.BodySchema{
+					Attributes: []hclext.AttributeSchema{
+						{
+							Name: providerTagsAttributeName,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	providerBody, err := runner.GetProviderContent("aws", providerSchema, nil)
+	if err != nil {
+		return nil
+	}
+
+	// Get provider default tags
+	var providerAlias string
+	for _, provider := range providerBody.Blocks.OfType("provider") {
+		// Get the alias attribute, in terraform when there is a single aws provider its called "default"
+		providerAttr, ok := provider.Body.Attributes["alias"]
+		if !ok {
+			providerAlias = "default"
+		} else {
+			err := runner.EvaluateExpr(providerAttr.Expr, func(alias string) error {
+				providerAlias = alias
+				return nil
+			}, nil)
+			if err != nil {
+				return nil
+			}
+		}
+		logger.Debug("Walk `%s` provider", providerAlias)
+
+		if len(provider.Body.Blocks) == 0 {
+			if err := runner.EmitIssue(r, fmt.Sprintf("The provider `%s` is missing the `default_tags` block", providerAlias), provider.DefRange); err != nil {
+				return err
+			}
+			continue
+		}
+
+		for _, block := range provider.Body.Blocks {
+			var providerTags []string
+			attr, ok := block.Body.Attributes[providerTagsAttributeName]
+			if !ok {
+				r.emitIssue(runner, providerTags, config, provider.DefRange)
+				continue
+			}
+
+			err := runner.EvaluateExpr(attr.Expr, func(val cty.Value) error {
+				keys, known := getKeysForValue(val)
+
+				if !known {
+					logger.Warn("The missing aws tags rule can only evaluate provided variables, skipping %s.", provider.Labels[0]+"."+providerAlias+"."+providerDefaultTagsBlockName+"."+providerTagsAttributeName)
+					return nil
+				}
+
+				logger.Debug("Walk `%s` provider with tags `%v`", providerAlias, keys)
+				providerTags = keys
+				return nil
+			}, nil)
+
+			if err != nil {
+				return nil
+			}
+
+			// Check tags
+			r.emitIssue(runner, providerTags, config, provider.DefRange)
```

Comment:
`attr.Range` would be better than `provider.DefRange`.

Reply:
changed